### ISSUE TITLE
[Bug] Shifted tables in DeadOil_PhaseModel

### ISSUE THREAD

**Issue by francoishamon** [GEOSX/PVTPackage#29]
In `DeadOil_PhaseModel.cpp`, the formation volume factors and viscosities are interpolated from a table as a function of pressure. In the current implementation, we add one pressure value (the surface pressure) to the tabulated pressures provided by the user here:
https://github.com/GEOSX/PVTPackage/blob/a367be7eb9364a62d0a3e13d9a7e84cad48b4775/PVTPackage/source/MultiphaseSystem/PhaseModel/BlackOil/DeadOil_PhaseModel.cpp#L97-L103
However, we do not add a value for the tabulated formation volume factors and the viscosities, so the table is now shifted by one. One can see that here:
https://github.com/GEOSX/PVTPackage/blob/a367be7eb9364a62d0a3e13d9a7e84cad48b4775/PVTPackage/source/MultiphaseSystem/PhaseModel/BlackOil/DeadOil_PhaseModel.cpp#L139-L141
by printing the indices `i_lower_branch` and `i_upper_branch` and the values of the tabulated pressures, formation volume factors, and viscosities at these indices.

@TotoGaz if that is ok with you, I am going to remove the piece of code that inserts the atmospheric pressure into the table. It will be the responsibility of the user to provide a table that covers the range of the values encountered in the simulation (if this is not the case GEOSX will throw an error). This change will require a rebaseline of the integrated tests, which I can do as part of GEOSX PR  https://github.com/GEOSX/GEOSX/pull/1155.

**Comment by TotoGaz** [comment]
@francoishamon go ahead, please keep patch as minimal as possible (no fancy renaming, no indentation corrections...), so I can pick it the most easy way.

**Comment by TotoGaz** [comment]
@francoishamon Will you have to update some `integratedTests`? Don't worry about unit tests 🤣

**Comment by francoishamon** [comment]
Yes. I have not checked but I assume that it will change the results of the dead-oil integrated tests in the main repository. I have a PR (https://github.com/GEOSX/GEOSX/pull/1155) that is ready and will require a rebaseline  (of the same integrated tests, because I registered some new variables) anyway, so my plan was to update PVTPackage and the integrated tests as part of https://github.com/GEOSX/GEOSX/pull/1155.

**Comment by TotoGaz** [comment]
Good, just wanted to be sure you were aware of that.
